packages/lib-classifier/src/store/SubjectStore/SingleTextSubject/SingleTextSubject.js

```diff
@@ -1,11 +1,65 @@
-import { types } from 'mobx-state-tree'
-import createLocationCounts from '@helpers/createLocationCounts'
+import { flow, types } from 'mobx-state-tree'
+import request from 'superagent'
+import asyncStates from '@zooniverse/async-states'
+
+import { createLocationCounts } from '@helpers'
 import Subject from '../Subject'
 
-const SingleTextSubject = types
-  .refinement('SingleTextSubject', Subject, subject => {
-    const counts = createLocationCounts(subject)
-    return subject.locations.length === 1 && counts.text === 1
+const TextSubject = types
+  .model('TextSubject', {
+    content: types.maybeNull(types.string),
+    contentLoadingState: types.optional(types.enumeration('contentLoadingState', asyncStates.values), asyncStates.initialized),
+    error: types.maybeNull(types.frozen({}))
   })
+  .actions(self => {
+    function afterAttach () {
+      if (self.contentLoadingState === asyncStates.initialized) {
+        self.fetchContent()
+      }
+    }
+
+    function getSubjectUrl () {
+      // Find locations that have a text/plain MIME type.
+      const textLocation = self.locations.find(l => l['text/plain']) || {}
+      const url = Object.values(textLocation)[0]
+      if (url) {
+        return url
+      } else {
+        throw new Error('No text url found for this subject')
+      }
+    }
+
+    function * fetchContent () {
+      self.contentLoadingState = asyncStates.loading
+
+      try {
+        const url = getSubjectUrl()
+        const response = yield request.get(url)
+
+        if (response.text) {
+          self.content = response.text
+        }
+
+        self.contentLoadingState = asyncStates.success
+      } catch (error) {
+        self.contentLoadingState = asyncStates.error
+        self.error = error
+      }
+    }
+
+    return {
+      afterAttach,
+      fetchContent: flow(fetchContent)
+    }
+  })
+
+const SingleTextSubject = types
+  .refinement(
+    'SingleTextSubject',
+    types.compose('SingleTextSubject', Subject, TextSubject),
+    subject => {
+      const counts = createLocationCounts(subject)
+      return subject.locations.length === 1 && counts.text === 1
```

Comment:
This filters out PH-TESS subjects, right?

Reply:
Out of interest, is there a subject viewer configured in Talk for plain text subjects?

Reply:
yes this refinement would filter out the TESS subjects (usually `.png` and `.txt`).

Here's the PFE text viewer - https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/components/file-viewer/text-viewer.jsx